geode-tcp-server/src/main/java/org/apache/geode/distributed/internal/tcpserver/LocatorAddress.java

```diff
@@ -15,96 +15,77 @@
 package org.apache.geode.distributed.internal.tcpserver;
 
 import java.net.InetSocketAddress;
+import java.util.Objects;
 
 import org.apache.commons.validator.routines.InetAddressValidator;
 
 public class LocatorAddress {
 
   private final InetSocketAddress socketInetAddress;
-  private final String hostname;
-  private final int port;
-  private final boolean isIpString;
 
   public LocatorAddress(InetSocketAddress loc, String locStr) {
-    this.socketInetAddress = loc;
-    this.hostname = locStr;
-    this.port = loc.getPort();
-    this.isIpString = InetAddressValidator.getInstance().isValid(locStr);
+    if (InetAddressValidator.getInstance().isValid(locStr)) {
```

Comment:
I think we could add some unit tests for LocatorAddress to cover some of these logic changes.  I'm thinking things like the following pseudo code

```java
public class LocatorAddressTest {
    @Test
    public void getSocketInentAddress_whenResolvableAddress_returnsTheSocketAddress() {
        InetSocketAddress thing = new InetSocketAddress(8080);
        LocatorAddress subject = new LocatorAddress(thing, "localhost");

        InetSocketAddress actual = subject.getSocketInetAddress();

        assertThat(actual, isEqualTo(thing));
        assertThat(!actual.isUnresolved());
    }
}
```

Not 100% sure it will work out the way I think, but may be worth a shot